### PR TITLE
thrift: add components

### LINKED ISSUE #2655: [package] arrow/1.0.0: doesn't compile with parquet=True
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **arrow/1.0.0**
  * Operating System+version: **Linux Ubuntu 18.04**
  * Compiler+version: **GCC 8.4.0**
  * Conan version: **conan 1.28.1**
  * Python version: **Python 3.6.9**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
Configuration for profile default:

[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=8
compiler.libcxx=libstdc++11
build_type=Release
[options]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)
Trying to add arrow dependency with parquet support, in my conanfile.txt:
```
[requires]
arrow/1.0.0

[generators]
cmake

[options]
arrow:parquet=True

[imports]
include,* -> ./include
lib,* -> ./lib
bin,* -> ./bin
```
Thrift cannot be found by cmake:
```
-- Checking for module 'thrift'                                                                                                                                                          
--   No package 'thrift' found                                                                                                                                                           
CMake Error at /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:164 (message):                                                                                          
  Could NOT find Thrift (missing: THRIFT_STATIC_LIB) (found suitable version                                                                                                             
  "0.13.0", minimum required is "0.11.0")
```

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

cmake logs:
```
-- Conan: checking conan executable
-- Conan: Found program /usr/local/bin/conan
-- Conan: Version found Conan version 1.28.1

-- Conan: Automatic detection of conan settings from cmake
-- Conan: Settings= -s;build_type=Debug;-s;compiler=gcc;-s;compiler.version=8;-s;compiler.libcxx=libstdc++11
-- Conan: checking conan executable
-- Conan: Found program /usr/local/bin/conan
-- Conan: Version found Conan version 1.28.1

-- Conan executing: /usr/local/bin/conan install /home/ms/dev/test_project/conanfile.txt -s build_type=Debug -s compiler=gcc -s compiler.version=8 -s compiler.libcxx=libstdc++11 -g=cmake --build=missing
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=8
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

conanfile.txt: Installing package
Requirements
    arrow/1.0.0 from 'conan-center' - Cache
    boost/1.73.0 from 'conan-center' - Cache
    bzip2/1.0.8 from 'conan-center' - Cache
    libevent/2.1.11 from 'conan-center' - Cache
    openssl/1.1.1g from 'conan-center' - Cache
    thrift/0.13.0 from 'conan-center' - Cache
    zlib/1.2.11 from 'conan-center' - Cache
Packages
    arrow/1.0.0:c26df1c75dece9af94f65101d5889aff3f16bc48 - Build
    boost/1.73.0:5055c0c4fc0c2164e2d11dde7ef188260ed3ce3a - Cache
    bzip2/1.0.8:d666801fbcdc3fd37cf8b2a827d72e84b1878ff5 - Cache
    libevent/2.1.11:83c8982ab18513337f2443e7a495baa4bda7ad12 - Cache
    openssl/1.1.1g:761432f3453a17c960259b5b2605153e3bcffd7d - Cache
    thrift/0.13.0:cc79aaa29cb36348c899a8f538ff485dbf9c2481 - Cache
    zlib/1.2.11:761432f3453a17c960259b5b2605153e3bcffd7d - Cache

Installing (downloading, building) binaries...
bzip2/1.0.8: Already installed!
openssl/1.1.1g: Already installed!
zlib/1.2.11: Already installed!
boost/1.73.0: Already installed!
boost/1.73.0: LIBRARIES: ['boost_wave', 'boost_container', 'boost_contract', 'boost_exception', 'boost_graph', 'boost_iostreams', 'boost_locale', 'boost_log', 'boost_program_options', 'boost_random', 'boost_regex', 'boost_serialization', 'boost_wserialization', 'boost_coroutine', 'boost_fiber', 'boost_context', 'boost_timer', 'boost_thread', 'boost_chrono', 'boost_date_time', 'boost_atomic', 'boost_filesystem', 'boost_system', 'boost_type_erasure', 'boost_log_setup', 'boost_math_c99', 'boost_math_c99f', 'boost_math_c99l', 'boost_math_tr1', 'boost_math_tr1f', 'boost_math_tr1l', 'boost_nowide', 'boost_stacktrace_addr2line', 'boost_stacktrace_backtrace', 'boost_stacktrace_basic', 'boost_stacktrace_noop', 'boost_unit_test_framework']
boost/1.73.0: Package folder: /home/ms/.conan/data/boost/1.73.0/_/_/package/5055c0c4fc0c2164e2d11dde7ef188260ed3ce3a
libevent/2.1.11: Already installed!
thrift/0.13.0: Already installed!
thrift/0.13.0: Appending PATH env var with : /home/ms/.conan/data/thrift/0.13.0/_/_/package/cc79aaa29cb36348c899a8f538ff485dbf9c2481/bin
arrow/1.0.0: Copying sources to build folder
arrow/1.0.0: Building your package in /home/ms/.conan/data/arrow/1.0.0/_/_/build/c26df1c75dece9af94f65101d5889aff3f16bc48
arrow/1.0.0: Generator cmake_find_package created Findthrift.cmake
arrow/1.0.0: Generator cmake_find_package created FindBoost.cmake
arrow/1.0.0: Generator cmake_find_package created FindLibevent.cmake
arrow/1.0.0: Generator cmake_find_package created FindZLIB.cmake
arrow/1.0.0: Generator cmake_find_package created FindBZip2.cmake
arrow/1.0.0: Generator cmake_find_package created FindOpenSSL.cmake
arrow/1.0.0: Generator cmake created conanbuildinfo.cmake
arrow/1.0.0: Calling build()
-- The C compiler identification is GNU 8.4.0
-- The CXX compiler identification is GNU 8.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- Conan: Adjusting fPIC flag (ON)
-- Conan: Compiler GCC>=5, checking major version 8
-- Conan: Checking correct version: 8
-- Conan: C++ stdlib: libstdc++11
-- Building using CMake version: 3.17.2
-- Arrow version: 1.0.0 (full: '1.0.0')
-- Arrow SO version: 100 (full: 100.0.0)
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1")
-- clang-tidy not found
-- clang-format not found
-- Could NOT find ClangTools (missing: CLANG_FORMAT_BIN CLANG_TIDY_BIN)
-- infer not found
fatal: not a git repository (or any parent up to mount point /home/ms)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
-- Found Python3: /usr/bin/python3.6 (found version "3.6.9") found components: Interpreter
-- Using ccache: /usr/local/bin/ccache
-- Found cpplint executable at CPPLINT_BIN-NOTFOUND
-- System processor: x86_64
-- Performing Test CXX_SUPPORTS_SSE4_2
-- Performing Test CXX_SUPPORTS_SSE4_2 - Success
-- Performing Test CXX_SUPPORTS_AVX2
-- Performing Test CXX_SUPPORTS_AVX2 - Success
-- Performing Test CXX_SUPPORTS_AVX512
-- Performing Test CXX_SUPPORTS_AVX512 - Success
-- Arrow build warning level: CHECKIN
Using ld linker
Configured for DEBUG build (set with cmake -DCMAKE_BUILD_TYPE={release,debug,...})
-- Build Type: DEBUG
-- Using SYSTEM approach to find dependencies
-- ARROW_ABSL_BUILD_VERSION: 2eba343b51e0923cd3fb919a6abd6120590fc059
-- ARROW_AWSSDK_BUILD_VERSION: 1.7.160
-- ARROW_BOOST_BUILD_VERSION: 1.71.0
-- ARROW_BROTLI_BUILD_VERSION: v1.0.7
-- ARROW_BZIP2_BUILD_VERSION: 1.0.8
-- ARROW_CARES_BUILD_VERSION: 1.16.1
-- ARROW_GBENCHMARK_BUILD_VERSION: v1.5.1
-- ARROW_GFLAGS_BUILD_VERSION: v2.2.2
-- ARROW_GLOG_BUILD_VERSION: v0.4.0
-- ARROW_GRPC_BUILD_VERSION: v1.29.1
-- ARROW_GTEST_BUILD_VERSION: 1.8.1
-- ARROW_JEMALLOC_BUILD_VERSION: 5.2.1
-- ARROW_LZ4_BUILD_VERSION: v1.9.2
-- ARROW_MIMALLOC_BUILD_VERSION: v1.6.3
-- ARROW_ORC_BUILD_VERSION: 1.6.2
-- ARROW_PROTOBUF_BUILD_VERSION: v3.12.1
-- ARROW_RAPIDJSON_BUILD_VERSION: 1a803826f1197b5e30703afe4b9c0e7dd48074f5
-- ARROW_RE2_BUILD_VERSION: 2019-08-01
-- ARROW_SNAPPY_BUILD_VERSION: 1.1.8
-- ARROW_THRIFT_BUILD_VERSION: 0.12.0
-- ARROW_THRIFT_BUILD_MD5_CHECKSUM: 3deebbb4d1ca77dd9c9e009a1ea02183
-- ARROW_UTF8PROC_BUILD_VERSION: v2.5.0
-- ARROW_ZLIB_BUILD_VERSION: 1.2.11
-- ARROW_ZSTD_BUILD_VERSION: v1.4.5
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE
-- Building without OpenSSL support. Minimum OpenSSL version 1.0.2 required.
-- Checking for module 'thrift'
--   No package 'thrift' found
CMake Error at /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
  Could NOT find Thrift (missing: THRIFT_STATIC_LIB) (found suitable version
  "0.13.0", minimum required is "0.11.0")
Call Stack (most recent call first):
  /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:445 (_FPHSA_FAILURE_MESSAGE)
  source_subfolder/cpp/cmake_modules/FindThrift.cmake:87 (find_package_handle_standard_args)
  source_subfolder/cpp/cmake_modules/ThirdpartyToolchain.cmake:178 (find_package)
  source_subfolder/cpp/cmake_modules/ThirdpartyToolchain.cmake:1242 (resolve_dependency_with_version)
  source_subfolder/cpp/CMakeLists.txt:484 (include)


-- Configuring incomplete, errors occurred!
See also "/home/ms/.conan/data/arrow/1.0.0/_/_/build/c26df1c75dece9af94f65101d5889aff3f16bc48/CMakeFiles/CMakeOutput.log".
See also "/home/ms/.conan/data/arrow/1.0.0/_/_/build/c26df1c75dece9af94f65101d5889aff3f16bc48/CMakeFiles/CMakeError.log".
arrow/1.0.0:
arrow/1.0.0: WARN: Package is corrupted, removing folder: /home/ms/.conan/data/arrow/1.0.0/_/_/package/c26df1c75dece9af94f65101d5889aff3f16bc48
arrow/1.0.0: WARN: Build folder is dirty, removing it: /home/ms/.conan/data/arrow/1.0.0/_/_/build/c26df1c75dece9af94f65101d5889aff3f16bc48
arrow/1.0.0: ERROR: Package 'c26df1c75dece9af94f65101d5889aff3f16bc48' build failed
arrow/1.0.0: WARN: Build folder /home/ms/.conan/data/arrow/1.0.0/_/_/build/c26df1c75dece9af94f65101d5889aff3f16bc48
ERROR: arrow/1.0.0: Error in build() method, line 351
	cmake = self._configure_cmake()
while calling '_configure_cmake', line 338
	self._cmake.configure()
	ConanException: Error 1 while executing cd '/home/ms/.conan/data/arrow/1.0.0/_/_/build/c26df1c75dece9af94f65101d5889aff3f16bc48' && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE="Debug" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="gcc" -DCONAN_COMPILER_VERSION="8" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DCONAN_LIBCXX="libstdc++11" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="/home/ms/.conan/data/arrow/1.0.0/_/_/package/c26df1c75dece9af94f65101d5889aff3f16bc48" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCONAN_CMAKE_POSITION_INDEPENDENT_CODE="ON" -DCMAKE_MODULE_PATH="/home/ms/.conan/data/arrow/1.0.0/_/_/build/c26df1c75dece9af94f65101d5889aff3f16bc48" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -DARROW_DEPENDENCY_SOURCE="SYSTEM" -DARROW_GANDIVA="False" -DARROW_PARQUET="True" -DARROW_PLASMA="False" -DARROW_DATASET="False" -DARROW_FILESYSTEM="False" -DPARQUET_REQUIRE_ENCRYPTION="False" -DARROW_HDFS="False" -DARROW_VERBOSE_THIRDPARTY_BUILD="True" -DARROW_BUILD_SHARED="False" -DARROW_BUILD_STATIC="True" -DARROW_NO_DEPRECATED_API="False" -DARROW_FLIGHT="False" -DARROW_HIVESERVER2="False" -DARROW_COMPUTE="False" -DARROW_CSV="False" -DARROW_CUDA="False" -DARROW_JEMALLOC="False" -DARROW_JSON="False" -DBOOST_SOURCE="SYSTEM" -DProtobuf_SOURCE="SYSTEM" -DgRPC_SOURCE="SYSTEM" -DARROW_USE_GLOG="False" -DGLOG_SOURCE="SYSTEM" -DARROW_WITH_BACKTRACE="False" -DARROW_WITH_BROTLI="False" -DBrotli_SOURCE="SYSTEM" -DARROW_BROTLI_USE_SHARED="False" -Dgflags_SOURCE="SYSTEM" -DARROW_WITH_BZ2="False" -DBZip2_SOURCE="SYSTEM" -DARROW_WITH_LZ4="False" -DLz4_SOURCE="SYSTEM" -DARROW_WITH_SNAPPY="False" -DSnappy_SOURCE="SYSTEM" -DARROW_WITH_ZLIB="False" -DRE2_SOURCE="SYSTEM" -DZLIB_SOURCE="SYSTEM" -DARROW_WITH_ZSTD="False" -DZSTD_SOURCE="SYSTEM" -DORC_SOURCE="SYSTEM" -DARROW_WITH_THRIFT="True" -DThrift_SOURCE="SYSTEM" -DTHRIFT_VERSION="1.0" -DARROW_USE_OPENSSL="False" -DARROW_S3="False" -DAWSSDK_SOURCE="SYSTEM" -DARROW_BUILD_UTILITIES="False" -DARROW_BUILD_INTEGRATION="False" -DARROW_INSTALL_NAME_RPATH="False" -DARROW_BUILD_EXAMPLES="False" -DARROW_BUILD_TESTS="False" -DARROW_ENABLE_TIMING_TESTS="False" -DARROW_BUILD_BENCHMARKS="False" -DLLVM_SOURCE="SYSTEM" -DARROW_WITH_UTF8PROC="False" -Dutf8proc_SOURCE="SYSTEM" -Wno-dev '/home/ms/.conan/data/arrow/1.0.0/_/_/build/c26df1c75dece9af94f65101d5889aff3f16bc48'
CMake Error at build/conan.cmake:402 (message):
  Conan install failed='1'
Call Stack (most recent call first):
  build/conan.cmake:497 (conan_cmake_install)
  CMakeLists.txt:79 (conan_cmake_run)


-- Configuring incomplete, errors occurred!
See also "/home/ms/dev/test_project/build/CMakeFiles/CMakeOutput.log".
See also "/home/ms/dev/test_project/build/CMakeFiles/CMakeError.log".

```

</details>

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **thrift/0.13.0**

Fixes #2655

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by madebr** [comment]
@uilianries @danimtb 
This error appears again.
```
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppBuild.targets(321,5): error MSB3491: Could not write lines to file "thrift-compiler.dir\Release\thrift-compiler.tlog\thrift-compiler.lastbuildstate". The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters. [C:\J\w\cci_PR-2664@2\2\b965cdf8-8e06-4e3a-9be0-9684b440f3dd\.conan\data\thrift\0.13.0\_\_\build\16fa88dc24ae68d4dc0d3d93b5dca02e4f4901da\build_subfolder\source_subfolder\compiler\cpp\thrift-compiler.vcxproj]
```

**Review by danimtb — DISMISSED** [review]

**Review by uilianries — DISMISSED** [review]

**Comment by madebr** [comment]
@SpaceIm @prince-chrismc @uilianries 
Can this pr get a review, please?

File: recipes/thrift/all/conanfile.py [hunk 1]
```diff
@@ -80,7 +85,11 @@ def build_requirements(self):
...
     def requirements(self):
-        self.requires("boost/1.73.0")
+        self.requires("boost/1.74.0")
+
+        if self.options.with_qt:
+            # FIXME: missing qt recipe
+            raise ConanInvalidConfiguration("qt is not (yet) available on cci")
 
         if self.options.with_openssl:
             self.requires("openssl/1.1.1g")
```
Comment (prince-chrismc) [inline recipes/thrift/all/conanfile.py:95]:
```suggestion
            self.requires("openssl/1.1.1h")
```

File: recipes/thrift/all/conanfile.py [hunk 2]
```diff
@@ -130,17 +151,53 @@ def package(self):
...
+        self.cpp_info.names["cmake_find_package_multi"] = "thrift"
+
+        self.cpp_info.components["libthrift"].libs = ["thrift" + libsuffix]
+        self.cpp_info.components["libthrift"].names["cmake_find_package"] = "thrift"
+        self.cpp_info.components["libthrift"].names["cmake_find_package_multi"] = "thrift"
+        self.cpp_info.components["libthrift"].names["pkg_config"] = "thrift"
         if self.settings.os == "Windows":
-            # To avoid error C2589: '(' : illegal token on right side of '::'
-            self.cpp_info.defines.append("NOMINMAX")
+            # To avoid error C2589: "(" : illegal token on right side of "::"
```
Comment (prince-chrismc) [inline recipes/thrift/all/conanfile.py:169]: This is pretty common
```suggestion
```

File: recipes/thrift/all/patches/cmake.diff [hunk 3]
```diff
@@ -5,15 +5,15 @@
  
  # If OpenSSL is not found or disabled just ignore the OpenSSL stuff
 -if(OPENSSL_FOUND AND WITH_OPENSSL)
-+if(OpenSSL_FOUND AND WITH_OPENSSL)
++if(WITH_OPENSSL)
```
Comment (prince-chrismc) [inline recipes/thrift/all/patches/cmake.diff:8]: I am surprised `OpenSSL_FOUND` is not set =?

File: recipes/thrift/all/conanfile.py [hunk 4]
```diff
@@ -130,17 +151,53 @@ def package(self):
...
         elif self.settings.os == "Linux":
-            self.cpp_info.system_libs.extend(["m", "pthread"])
+            self.cpp_info.components["libthrift"].system_libs.extend(["m", "pthread"])
+        self.cpp_info.components["libthrift"].requires.append("boost::boost")
+        if self.options.with_openssl:
+            self.cpp_info.components["libthrift"].requires.append("openssl::openssl")
+        if self.options.with_libevent:
+            self.cpp_info.components["libthrift"].requires.append("libevent::libevent")
+
+        # FIXME: this generates thrift::thriftz, but should be thriftz::thriftz
```
Comment (prince-chrismc) [inline recipes/thrift/all/conanfile.py:179]: Is there an issue for this in the conan repo?

File: recipes/thrift/all/patches/cmake.diff (hunk 3 above)
Comment (madebr) [reply recipes/thrift/all/patches/cmake.diff:8]: `OpenSSL_FOUND` is set, but not `OPENSSL_FOUND`.

File: recipes/thrift/all/conanfile.py (hunk 4 above)
Comment (madebr) [reply recipes/thrift/all/conanfile.py:179]: https://github.com/conan-io/conan/issues/7615
More or less.

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**